src/preprocess/mod.rs

```diff
@@ -34,5 +38,13 @@ pub trait Preprocessor {
 
     /// Run this `Preprocessor`, allowing it to update the book before it is
     /// given to a renderer.
-    fn run(&self, ctx: &PreprocessorContext, book: &mut Book) -> Result<()>;
+    fn run(&self, ctx: &PreprocessorContext, book: Book) -> Result<Book>;
+
+    /// A hint to `MDBook` whether this preprocessor is compatible with a
+    /// particular renderer.
+    /// 
+    /// By default, always returns `true`.
+    fn supports_renderer(&self, _renderer: &str) -> bool {
```

Comment:
Can it be this function is never called? (_I am using the webview to see all changed files at once, and this one doesn’t show up anywhere else_)

Reply:
Ah, maybe it’s still a WIP, and it’s more about comments on the general direction :). Never mind.

Reply:
Yeah this PR is still a work in progress. Eventually I want to extract the logic out into a `should_run()` function which will check the config if the user explicitly overrides what preprocessors are run, then fall back to asking a preprocessor directly whether it supports a specific renderer.